### PR TITLE
[FLINK-19811][table-planner-blink] Simplify SEARCHes in FlinkRexUtil#simplify

### PR DESCRIPTION
**Pull request by TsReaper**
## What is the purpose of the change

The new version of Calcite introduces the `SEARCH` rex call to express range conditions. However `SEARCH`es in conjunctions are currently not simplified. For example, `AND(=($2, 2020), SEARCH($2, Sarg[2020, 2021]))` is not simplified while it should be simplified to `=($2, 2020)`.

This PR expands the `SEARCH` rex calls in `FlinkRexUtil#simplify` to achieve this. `RexSimplify#simplify` in Calcite will do the rest of the work.

## Brief change log

 - Simplify `SEARCH`es in conjunctions in `FlinkRexUtil#simplify`

## Verifying this change

This change added tests and can be verified by running the added tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala [hunk 1]
```diff
@@ -196,14 +196,23 @@ object FlinkRexUtil {
     * 4. (a > b OR c < 10) OR b < a -> a > b OR c < 10
     * 5. a = a, a >= a, a <= a -> true
     * 6. a <> a, a > a, a < a -> false
+    * 7. a = 2020 AND SEARCH(a, [2020, 2021]) -> a = 2020
```
Comment (godfreyhe) [inline flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala:199]: does `a = 2020 OR SEARCH(a, [2020, 2021])` can be simplified as `a = 2020`  ?

Comment (TsReaper) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala:199]: Obviously it can't.

Comment (TsReaper) [reply flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/plan/utils/FlinkRexUtil.scala:199]: This should be simplified as `SEARCH(a, [2020, 2021])`.

**Review by godfreyhe — APPROVED** [review]
LGTM

**Comment by godfreyhe** [comment]
update the title as "Simplify SEARCHes in FlinkRexUtil#simplify"

**Comment by godfreyhe** [comment]
replaced by https://github.com/apache/flink/pull/13931

**Pull request closed without merging**